src/algebra/ordered_ring.lean

```diff
@@ -753,6 +781,10 @@ addition are monotone. -/
 @[priority 100] -- see Note [lower instance priority]
 instance decidable_linear_ordered_comm_ring.to_decidable_linear_ordered_semiring [d : decidable_linear_ordered_comm_ring α] :
    decidable_linear_ordered_semiring α :=
+-- One might hope that `{ ..linear_ordered_ring.to_linear_ordered_semiring, ..d }`
+-- achieved the same result here.
+-- Unfortunately with that definition we see mismatched `preorder ℝ` instances in
+-- `topology.metric_space.basic`.
```

Comment:
So is there a potential diamond here?

Reply:
I'm not too sure what is going on here. I tried to do the "cleanup" step of using `{ ..linear_ordered_ring.to_linear_ordered_semiring, ..d }` instead, only to find a mysterious failure in `topology.metric_space.basic`.